packages/vite/src/node/cli.ts

```diff
@@ -50,6 +52,27 @@ function cleanOptions(options: GlobalCLIOptions) {
   return ret
 }
 
+function bindShortcut(server: ViteDevServer) {
+  process.stdin.resume()
+  process.stdin.setEncoding('utf8')
+  process.stdin.on('data', async (data) => {
+    const str = data.toString().trim().toLowerCase()
+    switch (str) {
+      case 'r':
+        await restartServer(server)
```

Comment:
Can this promise be rejected? If so, we should handle it with try-catch.

Reply:
no need I think https://github.com/vitejs/vite/blob/a30724c8a492016a3f4fd1b209975b49e6972135/packages/vite/src/node/server/hmr.ts#L62